enhancements/sdk-kubernetes-1.21.md

```diff
@@ -0,0 +1,219 @@
+- Making any necessary changes to get CI to pass.
+
+#### Upgrade operator-framework/operator-lib to support 1.21
+
+_GitHub: [operator-framework/operator-lib][gh-operator-lib]_
+
+##### Graduation Criteria
+
+- Bumping the controller-runtime version in go.mod scaffolds to the controller-runtime version that supports Kubernetes 1.21.
+- Updating e2e tests to use Kubernetes 1.21 where possible and create follow-on tasks for tests that cannot be updated (this often happens when we update before kind images we use have been updated)
+- Making any necessary changes to get CI to pass.
+
+#### Upgrade operator-framework/api to support 1.21
+
+_GitHub: [operator-framework/api][gh-api]_
+
+##### Graduation Criteria
+
+- Bumping the controller-runtime version in go.mod scaffolds to the controller-runtime version that supports Kubernetes 1.21.
+- Updating tests to use Kubernetes 1.21 where possible and create follow-on tasks for tests that cannot be updated (this often happens when we update before kind images we use have been updated)
+- Making any necessary changes to get CI to pass.
+
+#### Upgrade operator-framework/java-operator-plugins to support 1.21
+
+_GitHub: [operator-framework/java-operator-plugins][gh-java]_
+
+##### Graduation Criteria
+
+- Bumping the apimachinery and kubebuilder version in `go.mod` to a version that supports Kubernetes 1.21.
+- Updating e2e tests to use Kubernetes 1.21 where possible and create follow-on tasks for tests that cannot be updated (this often happens when we update before kind images we use have been updated)
+- Making any necessary changes to get CI to pass.
+
+[gh-controller-runtime]: https://github.com/kubernetes-sigs/controller-runtime
+[gh-controller-tools]: https://github.com/kubernetes-sigs/controller-tools
+[gh-kubebuilder]: https://github.com/kubernetes-sigs/kubebuilder
+[gh-operator-sdk]: https://github.com/operator-framework/operator-sdk
+[gh-operator-lib]: https://github.com/operator-framework/operator-lib
+[gh-api]: https://github.com/operator-framework/api
+[gh-java]: https://github.com/operator-framework/java-operator-plugins
+
+### Risks and Mitigations
+
+- As always there is risk to the actual development and delivery timing of this feature, due to the many upstream changes that are required to be completed before Operator SDK can support Kubernetes
+1.21.
+
+| Repositories where k8s is to bumped to 1.21        | Estimated timeline    |
+| -------------------------------------------------- | --------------------- |
+| Controller-runtime                                 | Done                  |
+| Controller-tools                                   | Done                  |
+| Kubebuilder binary, v3 plugin and kb-declarative pattern                   | ~ July 2              |
+| Operator-framework/api                             | ~ July 2              |
```

Comment:
I have estimated this based on our sprint timeline. This sprint we have brought in stories for KB, and next sprint we will have stories for bumping dependencies in SDK. That's why I have estimated it to be last date of Sprint 60 and 61.